cpp/src/shuffler/postbox.cpp

```diff
@@ -15,8 +15,14 @@ namespace rapidsmpf::shuffler::detail {
 template <typename KeyType>
 void PostBox<KeyType>::insert(Chunk&& chunk) {
     std::lock_guard const lock(mutex_);
-    auto [_, inserted] =
-        pigeonhole_[key_map_fn_(chunk.pid)].insert({chunk.cid, std::move(chunk)});
+    // TODO: this is a questionable change!
+    // now the outgoing_chunks_ postbox in shuffler is rank-based. If there are multiple
+    // messages in the chunk now, they SHOULD have the same target rank! Otherwise, we
+    // would have to slice the chunk, which is not what we want.
+    // This is why I think we should have a Chunk class and ChunkBatch class separately.
+    auto [_, inserted] = pigeonhole_[key_map_fn_(chunk.part_id(0))].insert(
+        {chunk.chunk_id(), std::move(chunk)}
+    );
```

Comment:
> If there are multiple messages in the chunk now, they SHOULD have the same target rank! 

But this is always true, right? I mean, we only concatenate chunks with the same target rank?